components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/event/queue/QueuedEvent.java

```diff
@@ -20,18 +20,25 @@
 
 import org.wso2.siddhi.core.event.Event;
 
+import java.util.Arrays;
+
 public class QueuedEvent {
     private String sourceHandlerElementId;
     private long sequenceID;
     private Event event;
     private long timestamp;
     private String siddhiAppName;
+    private String[] transportSyncProperties;
 
-    public QueuedEvent(String siddhiAppName, String sourceHandlerElementId, long sequenceID, Event event) {
+    public QueuedEvent(String siddhiAppName, String sourceHandlerElementId, long sequenceID, Event event,
+                       String[] transportSyncProperties) {
         this.sourceHandlerElementId = sourceHandlerElementId;
         this.siddhiAppName = siddhiAppName;
         this.sequenceID = sequenceID;
         this.event = event;
+        this.transportSyncProperties = transportSyncProperties != null ?
+                Arrays.copyOf(transportSyncProperties,transportSyncProperties.length) : null;
```

Comment:
Do we need to copy the properties here? I think we can omit that in favor of performance.

Reply:
Yes . In here findbugs complaining you need to handle the EI_EXPOSE_REP, EI_EXPOSE_REP2 errors thrown because "May expose internal representation by incorporating reference to mutable object. This code stores a reference to an externally mutable object into the internal representation of the object." Since in the code , no other code tries to change the object , i have suppressed the warning